### PR TITLE
feat: LT-1413 - Default Queue

### PR DESCRIPTION
**Pull request by andytaylorkainos**

### CODE REVIEW

File: assets/javascripts/filter-default-queue-list.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+let team = $("#team");
+
+function filterSelectOptions(team_value, default_to_select) {
+    // if team changes, set default_queue to 'select' and filter out other team queues
```
Comment (BenAndrew-Kainos) [inline assets/javascripts/filter-default-queue-list.js:4]: What is this used for?

File: core/helpers.py [hunk 2]
```diff
@@ -91,3 +91,25 @@ def convert_value_to_query_param(key: str, value):
...
+    Returns None if a key is not found instead of throwing a KeyError
+    Example usage:
+    get_nested_value(case, ['application', 'end_user', 'name'])
+    """
+
+    value = dictionary.get(nested_keys.pop(0), None)
+
+    if value:
+        while nested_keys:
+            next_value = value.get(nested_keys.pop(0), None)
```
Comment (BenAndrew-Kainos) [inline core/helpers.py:108]: I'm very confused by this function. Does this not just return the last value that is None?

File: core/models.py [hunk 3]
```diff
@@ -43,6 +45,7 @@ class User(AbstractUser):
     email = models.EmailField(default=None, blank=True, unique=True)
     first_name = models.CharField(max_length=30, blank=True, default=None, null=True)
     last_name = models.CharField(max_length=150, blank=True, default=None, null=True)
+    default_queue = models.UUIDField(default=uuid.UUID(ALL_CASES_QUEUE_ID), null=False)
```
Comment (BenAndrew-Kainos) [inline core/models.py:48]: Why is this needed?

File: queues/views.py [hunk 4]
```diff
@@ -32,7 +32,7 @@ def get(self, request, **kwargs):
         case_type = request.GET.get("case_type")
         status = request.GET.get("status")
         sort = request.GET.get("sort")
-        queue_pk = kwargs["queue_pk"]
+        queue_pk = kwargs.get("queue_pk") or request.user.default_queue
```
Comment (BenAndrew-Kainos) [inline queues/views.py:35]: Shouldn't this have a default if no kwarg or default queue

File: users/forms/users.py [hunk 5]
```diff
@@ -24,8 +25,15 @@ def add_user_form(request):
...
             ),
+            Select(
+                title=AddUserForm.DefaultQueue.TITLE,
+                description=AddUserForm.DefaultQueue.DESCRIPTION,
+                name="default_queue",
+                options=get_queues(request, include_system=True, convert_to_options=True),
+            ),
         ],
         back_link=BackLink(AddUserForm.BACK_LINK, reverse_lazy("users:users")),
+        javascript_imports=["/assets/javascripts/filter-default-queue-list.js"],
```
Comment (BenAndrew-Kainos) [inline users/forms/users.py:36]: I don't understand why this needs javascript

File: core/models.py (hunk 3 above)
Comment (BenAndrew-Kainos) [reply core/models.py:48]: Nevermind, its for storing this locally right

File: core/helpers.py (hunk 2 above)
Comment (andytaylorkainos) [reply core/helpers.py:108]: removed entirely

File: core/models.py (hunk 3 above)
Comment (a-bek) [inline core/models.py:48]: How come we need to store this here + API?

Comment (andytaylorkainos) [reply core/models.py:48]: This is the way the frontends do "sessions"
